ocs_ci/deployment/cloud.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf8 -*-
+"""
+This module contains common code and a base class for any cloud platform
+deployment.
+"""
+
+import logging
+import os
+
+from ocs_ci.deployment.deployment import Deployment
+from ocs_ci.deployment.ocp import OCPDeployment as BaseOCPDeployment
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants, exceptions
+from ocs_ci.utility.bootstrap import gather_bootstrap
+from ocs_ci.utility.utils import get_cluster_name
+from ocs_ci.utility.utils import run_cmd
+
+
+logger = logging.getLogger(__name__)
+
+
+class CloudDeploymentBase(Deployment):
+    """
+    Base class for deployment on a cloud platform (such as AWS, Azure, ...).
+    """
+
+    def __init__(self):
+        """
+        Any cloud platform deployment requires region and cluster name.
+        """
+        super(CloudDeploymentBase, self).__init__()
+        self.region = config.ENV_DATA['region']
+        if config.ENV_DATA.get('cluster_name'):
+            self.cluster_name = config.ENV_DATA['cluster_name']
+        else:
+            self.cluster_name = get_cluster_name(self.cluster_path)
+
+
+class CloudIPIOCPDeployment(BaseOCPDeployment):
```

Comment:
+1 on creating this class.
I actually see that also vSphere IPI OCP deployment methods look similar to the ones here - https://github.com/red-hat-storage/ocs-ci/pull/2564/files#diff-9190895310341674a2b7ed1d4b07d5dfR388

How about calling this class `IPIOCPDeployment`?

Reply:
This is a good point. I will rename it in this pull request. Further simplification would need to wait for both PRs to be merged though.

Reply:
addressed in 6ba68e5